lib/medtronic600/NGPHistoryParser.js

```diff
@@ -0,0 +1,894 @@
+    const carbRatio = this.eventData.readUInt32BE(0x13);
+    return this.carbUnits === NGPUtil.NGPConstants.CARB_UNITS.GRAMS ?
+      carbRatio / 10.0 : carbRatio / 1000.0;
+  }
+
+  get isf() {
+    const isf = this.eventData.readUInt16BE(0x11);
+    return this.bgUnits === NGPUtil.NGPConstants.BG_UNITS.MG_DL ? isf : isf / 10.0;
+  }
+
+  get lowBgTarget() {
+    const bgTarget = this.eventData.readUInt16BE(0x17);
+    return this.bgUnits === NGPUtil.NGPConstants.BG_UNITS.MG_DL ? bgTarget : bgTarget / 10.0;
+  }
+
+  get highBgTarget() {
+    const bgTarget = this.eventData.readUInt16BE(0x19);
+    return this.bgUnits === NGPUtil.NGPConstants.BG_UNITS.MG_DL ? bgTarget : bgTarget / 10.0;
+  }
+
+  get correctionEstimate() {
+    /* eslint-disable no-bitwise */
+    return ((this.eventData[0x1B] << 8) |
+      (this.eventData[0x1C] << 8) | (this.eventData[0x1D] << 8) | this.eventData[0x1E]) / 10000.0;
+    /* eslint-enable no-bitwise */
+  }
+
+  get foodEstimate() {
+    return this.eventData.readUInt32BE(0x1F) / 10000.0;
+  }
+
+  get iob() {
+    return this.eventData.readUInt32BE(0x23) / 10000.0;
+  }
+
+  get iobAdjustment() {
+    return this.eventData.readUInt32BE(0x27) / 10000.0;
+  }
+
+  get bolusWizardEstimate() {
+    return this.eventData.readUInt32BE(0x2B) / 10000.0;
+  }
+
+  get finalEstimate() {
+    return this.eventData.readUInt32BE(0x31) / 10000.0;
+  }
+
+  get estimateModifiedByUser() {
+    // eslint-disable-next-line no-bitwise
+    return (this.eventData.readUInt32BE(0x30) & 1) === 1;
+  }
+}
+
+class NGPHistoryParser {
+  constructor(cfg, pages) {
+    this.cfg = cfg;
+    this.events = [];
+
+    this.processPages(pages);
+  }
+
+  processPages(pages) {
+    // Consume the pages as we process them to save on memory.
+    const eventTypes = {}; // TODO - remove. Temporary while we're developing.
+    while (pages.length > 0) {
+      const page = pages.shift();
+      for (const event of NGPHistoryParser.eventsFromPageString(page)) {
+        if (eventTypes[event.eventType] === undefined) {
+          eventTypes[event.eventType] = 0;
+        }
+        this.events.push(event.eventInstance());
+        eventTypes[event.eventType] += 1;
+      }
+    }
+
+    debug('*** Processed event types:');
+    for (const key of Object.keys(eventTypes)) {
+      debug(`    ${key}:`, eventTypes[key]);
+    }
+  }
+
+  // eslint-disable-next-line generator-star-spacing, Beautify adds it, and it's not terrible
+  static * eventsFromPageString(pageString) {
+    const page = Buffer.from(pageString, 'hex');
+    let pos = 0;
+
+    while (pos < page.length) {
+      const eventSize = page[pos + 2];
+      const eventData = page.slice(pos, pos + eventSize);
+      pos += eventSize;
+      yield new NGPHistoryEvent(eventData);
+    }
+  }
+
+  * eventsOfType(eventType) {
+    for (const event of this.events) {
+      if (event.eventType === eventType) {
+        yield event;
+      }
+    }
+  }
+
+  findSuppressedBasal(tempBasal) {
```

Comment:
Oh, and I don't know if you've seen it in the data model docs yet, but suppressed info can be nested as well :/